app/email/email_batching.go

```diff
@@ -148,7 +159,34 @@ func (job *EmailBatchingJob) handleNewNotifications() {
 
 func (job *EmailBatchingJob) checkPendingNotifications(now time.Time, handler func(string, []*batchedNotification)) {
 	for userID, notifications := range job.pendingNotifications {
+		// Defensive code.
+		if len(notifications) == 0 {
+			continue
+		}
+
+		// get how long we need to wait to send notifications to the user
+		var interval int64
+		preference, err := job.service.store.Preference().Get(userID, model.PreferenceCategoryNotifications, model.PreferenceNameEmailInterval)
```

Comment:
Unrelated to the changes but it seems like all this parsing logic would better fit into a separate function that can be more easily tested in isolation.

Reply:
Hmm .. I am 1/5 on keeping it as-is. `TestCheckPendingNotificationsCantParseInterval` achieves this by setting the DB directly. It's just couple of lines of code, but happy to change if you feel strongly.